engine/src/main/java/org/terasology/world/chunks/internal/ChunkImpl.java

```diff
@@ -446,6 +446,21 @@ public int hashCode() {
     }
 
     @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (!(obj instanceof ChunkImpl)) {
+            return false;
+        }
+
+        ChunkImpl that = (ChunkImpl) obj;
+
+        return Objects.equal(this.chunkPos, that.chunkPos);
```

Comment:
In case of another dimension, we surely should hash another value and also compare another value. But you are correct this could have unforeseen changes.

The important part of the `Object` contract here should be:

`hashCode()` must always return the same code for two objects which are equal according to the `equals()` implementation.

This is true here. I'll adjust the implementation so that `equals` calls only super and provide a comment why the super `equals()` implementation is correct in this case.